stacks/spica/src/bucket/pages/add/add.component.ts

```diff
@@ -91,13 +93,18 @@ export class AddComponent implements OnInit {
   }
 
   saveBucketRow() {
+    this.savingBucketState = true;
     if (!(this.data._schedule instanceof Date)) {
       delete this.data._schedule;
     }
 
     this.bds
       .replaceOne(this.bucketId, this.data)
       .toPromise()
-      .then(() => this.router.navigate(["bucket", this.bucketId]));
+      .then(data => {
+        this.savingBucketState = false;
+        this.router.navigate(["bucket/" + this.bucketId + "/" + data]);
```

Comment:
You should use this instead `this.router.navigate(["bucket", this.bucketId, data]);`. Way cleaner.